lib/datadog/tracing/contrib/active_storage/events/upload.rb

```diff
@@ -0,0 +1,58 @@
+# typed: false
+
+require 'datadog/tracing/metadata/ext'
+require 'datadog/tracing/contrib/active_storage/event'
+require 'datadog/tracing/contrib/active_storage/ext'
+require 'datadog/tracing/contrib/analytics'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module ActiveStorage
+        module Events
+          # Defines instrumentation for 'service_upload.active_storage' event.
+          #
+          # Uploaded a file to the remote blob service
+          module Upload
+            include ActiveStorage::Event
+
+            EVENT_NAME = 'service_upload.active_storage'.freeze
+
+            module_function
+
+            def event_name
+              self::EVENT_NAME
+            end
+
+            def span_name
+              Ext::SPAN_UPLOAD
+            end
+
+            def span_type
+              # Interacting with a cloud-based blob service via HTTP
+              Tracing::Metadata::Ext::HTTP::TYPE_OUTBOUND
+            end
+
+            def process(span, _event, _id, payload)
+              as_key = payload[:key]
+              as_service = payload[:service]
+              # checksum is also on payload but this has infinite cardinality and low value
+
+              span.service = configuration[:service_name] if configuration[:service_name]
+              span.resource = "#{as_service}: #{as_key}"
```

Comment:
I believe `as_key` has high cardinality, as it seems to be [unique per file/path stored](https://github.com/rails/rails/blob/5a5440ba727b5b9882ee49763f8d44ed4ffde017/activestorage/app/controllers/active_storage/direct_uploads_controller.rb#L8).
This falls under the same unresolved issue we have with instrumenting URL paths for arbitrary URLs: there's no reliable default quantizer we can think of that wouldn't cause high cardinality for some users.

I recommend the removal of `as_key` here, and instead set it as a tag (`span.set_tag(...,as_key)`).

Please let me know if my assumptions are incorrect and `as_key` is either a predictable value or can be quantized reliably.